### ISSUE TITLE
Deprecated ROOT.Long and ROOT.Double in ROOT 6.22

### ISSUE THREAD

**Issue by lucasflores** [HEPData/hepdata_lib#147]
Hello,

So I have been seeing this warning about ROOT.Double being deprecated in ROOT 6.22 for a while and successfully ignored it until now. In our latest software releases on ATLAS  we have finally moved to ROOT 6.22 and so these lines are finally broken https://github.com/HEPData/hepdata_lib/blob/master/hepdata_lib/root_utils.py#L410
The notes on this change in ROOT can be found here https://root.cern/doc/v622/release-notes.html#pyroot (or just search for ROOT.Double).
I was able to implement a fix in my version with the following changes in `root_utils.py` : 

```python
+ import ctypes
```
```python
L410| - x_val = r.Double()   L410| +  x_val = ctypes.c_double()
L411| - y_val = r.Double()   L411| +  y_val = ctypes.c_double()
```
 ```python
L413| - points["x"].append(float(x_val))   L413| + points["x"].append(float(x_val.value))
L414| - points["y"].append(float(x_val))   L414| + points["y"].append(float(x_val.value))
```

I can't say I have exhaustively tested this but doing a diff of  the `output/` HEPData directories (one using ROOT 6.20 with hepdatalib as is and one using ROOT 6.22 with my fix)  for my rather large and varied HEPData submission shows that their output is identical. 
I am not sure I have time these days to validate and merge this in in the nominal way myself but I thought I would bring it to your attention.

Best,
Lucas

**Comment by clelange** [comment]
Hi Lucas,

Thanks for pointing this out and providing a solution! We should soon have some personpower to help with the maintenance of hepdata_lib to get this implemented and validated. Do you happen to know if this is backward-compatible for all ROOT 6 versions?

Clemens

**Comment by lucasflores** [comment]
Hi Clemens,

Thanks for getting back tome, and yeah no problem. 

> Do you happen to know if this is backward-compatible for all ROOT 6 versions?

hmm I would not really know this :/  sorry, and I have not tested it on any earlier version than 6.20. However `ctypes` seems to be a pretty standard python library so I would think it should. It seems ROOT was just trying to move away from their own wheel inventions. 

And if your person power needs a next thing to do :) I also have in my fork here https://github.com/lucasflores/hepdata_lib implemented `THStack` and `THGraph2D (THGraph2DErrors)` functionality. I might have some more time soon myself to contribute more officially but I just thought I would share.

Best,
Lucas

**Comment by AndreasAlbert** [comment]
I believe this issue is now addressed by #149 
otherwise, feel free to repoen
